src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClient.cs

```diff
@@ -130,7 +130,7 @@ public PSRoleAssignment CreateRoleAssignment(FilterRoleAssignmentsOptions parame
         /// </summary>
         /// <param name="options">The filtering options</param>
         /// <returns>The filtered role assignments</returns>
-        public List<PSRoleAssignment> FilterRoleAssignments(FilterRoleAssignmentsOptions options)
+        public List<PSRoleAssignment> FilterRoleAssignments(FilterRoleAssignmentsOptions options, bool excludeAssignmentsForDeletedPrincipals = true)
```

Comment:
add this `excludeAssignmentsForDeletedPrincipals` to `FilterRoleAssignmentsOptions` class instead of a parameter.